android/src/com/mapswithme/maps/search/HotelsFilter.java

```diff
@@ -0,0 +1,90 @@
+package com.mapswithme.maps.search;
+
+import android.support.annotation.NonNull;
+
+public class HotelsFilter
+{
+  // *NOTE* keep this in sync with JNI counterpart.
+  public final static int TYPE_AND = 0;
+  public final static int TYPE_OR = 1;
+  public final static int TYPE_OP = 2;
+
+  public final int mType;
+
+  protected HotelsFilter(int type)
+  {
+    mType = type;
+  }
+
+  public static class And extends HotelsFilter
+  {
+    @NonNull public final HotelsFilter mLhs;
+    @NonNull public final HotelsFilter mRhs;
+
+    public And(@NonNull HotelsFilter lhs, @NonNull HotelsFilter rhs)
+    {
+      super(TYPE_AND);
+      mLhs = lhs;
+      mRhs = rhs;
+    }
+  }
+
+  public static class Or extends HotelsFilter
+  {
+    @NonNull public final HotelsFilter mLhs;
```

Comment:
Could you put nullness annotation above of field (on new string) please:

```
@NonNull
public final HotelsFilter mLhs;
```
? and do this for rest nullnes annotations in this file?